node/engine/engine.go

```diff
@@ -854,20 +850,7 @@ func (e *Engine) recordMessageMetrics(message protocols.Message) {
 
 func (e *Engine) checkError(err error) {
 	if err != nil {
-
 		e.logger.Err(err).Msgf("%s, error in run loop", e.store.GetAddress())
-
-		for _, nonFatalError := range nonFatalErrors {
-			if errors.Is(err, nonFatalError) {
-				return
-			}
-		}
```

Comment:
In my thinking, there is a class of irrecoverable errors for which the engine should panic (at least for now, while go-nitro is under rapid development). Panicking and exiting will help us diagnose unexpected nitro implementation issues.

There is a second class of non-fatal errors. These are errors a nitro node expects to encounter, and the node can continue to operate after encountering the error. Attempting to create a second ledger channel between a pair of participants is an example of a non-fatal error.